src/wallet/coin_selection.rs

```diff
@@ -110,16 +127,28 @@ pub type DefaultCoinSelectionAlgorithm = LargestFirstCoinSelection; // make the
 // prev_txid (32 bytes) + prev_vout (4 bytes) + sequence (4 bytes) + script_len (1 bytes)
 pub(crate) const TXIN_BASE_WEIGHT: usize = (32 + 4 + 4 + 1) * 4;
 
+pub type Waste = i64;
 /// Result of a successful coin selection
 #[derive(Debug)]
 pub struct CoinSelectionResult {
     /// List of outputs selected for use as inputs
     pub selected: Vec<Utxo>,
     /// Total fee amount in satoshi
     pub fee_amount: u64,
+    /// Waste value of current coin selection
+    waste: Cell<Option<Waste>>,
```

Comment:
The use of `get_waste` and the `waste: Cell<<Option<Waste>>` field it's
because I was assuming a worst case scenario where the algorithms doesn't
necessarily set the waste metric and when an external source requires the
metric they should have a way to fulfill the request.
To modify the field, as it was, Rust required the whole instance to be
mutable, something undesirable, as the selected coins shouldn't be modified
after the algorithm execution. So I decided to use Cell as a way to mimic
the private/public behavior that I know from other programming languages.

Taking the above and what you said in the comments: _"you just want to
calculate it once at the end of the coin selection"_. I think that it should
be correct to set the waste field of `CoinSelectionResult` as: `pub waste:
Waste` and, as you said, remove the `get_waste` method altogether.